SetupDataPkg/Tools/KnobService.py

```diff
@@ -800,10 +844,11 @@ def generate_cached_implementation(schema, header_path, efi_type=False):
                                 out.write(get_spacing_string(efi_type) + "}" + get_line_ending(efi_type))
                             if subknob.max != subknob.format.max:
                                 out.write(get_spacing_string(efi_type))
-                                out.write("if ({}->{} > KNOB__{}__MAX) {{".format(
+                                out.write("if ({}{}{}{} > KNOB__{}__MAX) {{".format(
+                                    "" if '.' in path else "*",
                                     naming_convention_filter("value", False, efi_type),
-                                    # don't take the '.'
-                                    path[1:],
+                                    "->" if '.' in path else "",
+                                    path[1:],  # don't take the '.'

```

Comment:
Same here

Reply:
See reply on other comment, resolving this to track in one place.